server/src/test/java/org/apache/druid/client/indexing/ClientCompactionRunnerInfoTest.java

```diff
@@ -27,117 +27,166 @@
+  public void testMSQEngineWithDynamicPartitionsSpecIsValid()
   {
     DataSourceCompactionConfig compactionConfig = createCompactionConfig(
         new DynamicPartitionsSpec(100, null),
         Collections.emptyMap(),
         null,
         null
     );
-    assertTrue(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
+    Assert.assertTrue(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
                                          .isValid());
   }
 
   @Test
-  public void testDimensionRangePartitionsSpecs()
+  public void testMSQEngineWithDimensionRangePartitionsSpecIsValid()
   {
     DataSourceCompactionConfig compactionConfig = createCompactionConfig(
         new DimensionRangePartitionsSpec(100, null, ImmutableList.of("partitionDim"), false),
         Collections.emptyMap(),
         null,
         null
     );
-    assertTrue(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
+    Assert.assertTrue(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
                                          .isValid());
   }
 
   @Test
-  public void testQueryGranularityAll()
+  public void testMSQEngineWithQueryGranularityAllIsValid()
   {
     DataSourceCompactionConfig compactionConfig = createCompactionConfig(
         new DynamicPartitionsSpec(3, null),
         Collections.emptyMap(),
         new UserCompactionTaskGranularityConfig(Granularities.ALL, Granularities.ALL, false),
         null
     );
-    assertTrue(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
+    Assert.assertTrue(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
                                           .isValid());
   }
 
   @Test
-  public void testRollupFalseWithMetricsSpec()
+  public void testMSQEngineWithRollupFalseWithMetricsSpecIsInValid()
   {
     DataSourceCompactionConfig compactionConfig = createCompactionConfig(
         new DynamicPartitionsSpec(3, null),
         Collections.emptyMap(),
         new UserCompactionTaskGranularityConfig(null, null, false),
         new AggregatorFactory[]{new LongSumAggregatorFactory("sum", "sum")}
     );
-    assertFalse(ClientCompactionRunnerInfo.validateCompactionConfig(compactionConfig, CompactionEngine.NATIVE)
-                                          .isValid());
+    CompactionConfigValidationResult validationResult = ClientCompactionRunnerInfo.validateCompactionConfig(
+        compactionConfig,
+        CompactionEngine.NATIVE
+    );
+    Assert.assertFalse(validationResult.isValid());
+    Assert.assertEquals(
+        "rollup in granularitySpec must be set to True if metricsSpec is specifed for MSQ engine.",
+        validationResult.getReason()
+    );
+  }
+
+  @Test
+  public void testMSQEngineWithUnsupportedMetricsSpecIsInValid()
```

Comment:
What exactly is the unsupported metrics spec? Either add a comment about it or rename the test method to indicate that.

```suggestion
  public void testMSQEngineWithUnsupportedMetricsSpecIsInvalid()
```

Reply:
Added a comment to avoid having too long a function name.